### PR TITLE
NATIVE-167: Add documentation for design-changes and persistence

### PR DESCRIPTION
**Pull request by maxammann**

### CODE REVIEW

File: docs/design-changes.md [hunk 1]
```diff
@@ -0,0 +1,9 @@
+# Design Changes
+
+###### Design Change: Release of the react-native app
+The design of the react-native app will differ in a few points.
+* "Extras" will be changed to "Angebote"
```
Comment (klinzo) [inline docs/design-changes.md]: Do we use translations of "Angebote" in other languages or will it still be "Extras" there?

File: docs/design-changes.md [hunk 2]
```diff
@@ -4,6 +4,21 @@
...
-* A Dashboard/Startpage will get added
-* If the user wants to change from a Page to an Event he has to navigate back to the Startpage
+* A Dashboard/Startpage is added
+ ![](figures/preview.jpg)
+ Note: The event preview is planned for a future design change.
+ Note: The header should contain the Integreat logo xor the back button on iOS and Android.
+  
+* If the user wants to change from a Page to an Event he has to navigate back to the Dashboard
 * A new Set of Icons will be defined and will replace all icons
+* name_without_prefix in Header und prefix in klammern dahinter
```
Comment (klinzo) [inline docs/design-changes.md]: Remove this

**Comment by maxammann** [comment]
Ready for review.

File: docs/01-debugging.md [hunk 3]
```diff
@@ -0,0 +1,25 @@
+# Debugging
+For debugging install [react-native-debugger](https://github.com/jhen0409/react-native-debugger)
+
+## Setup connectivity from device to metro
+On Android you have 2 options:
+* Setup a tunnel though adb: `adb reverse tcp:8081 tcp:8081`
+* Open the development options of the react-native app (shake device or Ctrl+M) and set a development host
```
Comment (NKlug) [inline docs/01-debugging.md]: Third:
* Attach the device with a cable   
:smile:

Comment (maxammann) [reply docs/01-debugging.md]: Just attaching it is not enough. You need to run adb reverse.

Comment (NKlug) [reply docs/01-debugging.md]: Maybe I don't quite understand what you mean by connectivity. I never executed any of the two options you listed.

File: README.md [hunk 4]
```diff
@@ -37,88 +30,41 @@ Note: All dependencies are handled by CocoaPods. The versions in node_modules/ s
...
-## Conventions
+### Installing libraries with native code
 
-For naming we follow the [airbnb style](https://github.com/airbnb/javascript/tree/master/react). 
-For the JavasScript code style we use the [standard style](https://standardjs.com/).
-For git commit messages [this style](https://github.com/erlang/otp/wiki/Writing-good-commit-messages).
+If you want to install an external library which brings native Android/iOS code you have to check several things:
+* Edit `Podfile` and `build.gradle` to include the native parts. DO NOT RUN `yarn react-native link`!
+* Run `pod update` in the `ios/` folder
+* Make sure the App still compiles on Android and iOS
```
Comment (Schedulaar) [inline README.md]:
```suggestion
* Make sure the app still compiles on Android and iOS.
```

File: docs/01-debugging.md [hunk 5]
```diff
@@ -0,0 +1,25 @@
+# Debugging
+For debugging install [react-native-debugger](https://github.com/jhen0409/react-native-debugger)
```
Comment (Schedulaar) [inline docs/01-debugging.md]:
```suggestion
For debugging install [react-native-debugger](https://github.com/jhen0409/react-native-debugger).
```

File: docs/01-debugging.md [hunk 6]
```diff
@@ -0,0 +1,25 @@
...
+
+## Setup connectivity from device to metro
+On Android you have 2 options:
+* Setup a tunnel though adb: `adb reverse tcp:8081 tcp:8081`
```
Comment (Schedulaar) [inline docs/01-debugging.md]:
```suggestion
* Setup a tunnel through adb: `adb reverse tcp:8081 tcp:8081`
```

File: docs/01-debugging.md (hunk 3 above)
Comment (Schedulaar) [reply docs/01-debugging.md]: Yep, connectivity is not a good word I guess... I have never used adb reverse as well, and was never asked for a development host (whatever that is).

File: docs/01-debugging.md [hunk 7]
```diff
@@ -0,0 +1,25 @@
...
+* Open the development options of the react-native app (shake device or Ctrl+M) and set a development host
+
+On iOS the post 8081 is tunneled by default. You can not change this. If the emulator is running on an other host you 
+can proxy the port with `yarn run proxy <host>`
```
Comment (Schedulaar) [inline docs/01-debugging.md]:
```suggestion
can proxy the port with `yarn run proxy <host>`.
```

File: docs/01-debugging.md [hunk 8]
```diff
@@ -0,0 +1,25 @@
+# Debugging
+For debugging install [react-native-debugger](https://github.com/jhen0409/react-native-debugger)
+
+## Setup connectivity from device to metro
+On Android you have 2 options:
+* Setup a tunnel though adb: `adb reverse tcp:8081 tcp:8081`
+* Open the development options of the react-native app (shake device or Ctrl+M) and set a development host
+
+On iOS the post 8081 is tunneled by default. You can not change this. If the emulator is running on an other host you
```
Comment (Schedulaar) [inline docs/01-debugging.md]:
```suggestion
On iOS the port 8081 is tunneled by default. You cannot change this. If the emulator is running on another host you 
```

File: docs/01-debugging.md [hunk 9]
```diff
@@ -0,0 +1,25 @@
...
+* Setup a tunnel though adb: `adb reverse tcp:8081 tcp:8081`
+* Open the development options of the react-native app (shake device or Ctrl+M) and set a development host
+
+On iOS the post 8081 is tunneled by default. You can not change this. If the emulator is running on an other host you 
+can proxy the port with `yarn run proxy <host>`
+
+## Steps to get started with debugging:
+* Start react-native-debugger
+* Start the metro bundler
+* Connect a real device or run an emulator though adb (e.g. `$ANDROID_HOME/emulator/emulator -avd <avd>`)
```
Comment (Schedulaar) [inline docs/01-debugging.md]:
```suggestion
* Connect a real device or run an emulator through adb (e.g. `$ANDROID_HOME/emulator/emulator -avd <avd>`)
```

File: docs/01-debugging.md [hunk 10]
```diff
@@ -0,0 +1,25 @@
...
+* Setup network connectivity from the emulator/device to the host where the bundler is running (see [here](#setup-connectivity-from-device-to-metro))
+* [Enter the development menu](#enter-the-development-menu) and enable "Remote JS Debugging"
+* The app should reload now and the debugger should connect to the device
+
+## Enter the development menu
+Shake the device or press Ctrl+M to enter the menu. If you are connected though adb to your android device you can also
```
Comment (Schedulaar) [inline docs/01-debugging.md]:
```suggestion
Shake the device or press Ctrl+M to enter the menu. If you are connected through adb to your android device you can also 
```

File: docs/02-common-problems.md [hunk 11]
```diff
@@ -0,0 +1,32 @@
+# If you encounter problems:
+
+## Errors when compiling double-conversion on iOS
+
+This happens when you use Xcode without running `yarn ios` or you did not install the pods.
+Try to reinstall the node_modules folder. The double-conversion library gets downloaded and installed in there when you run `yarn ios`.
+
+See [here](https://github.com/facebook/react-native/issues/21168#issuecomment-422700915) and [here](https://github.com/facebook/react-native/issues/20774) for more information
```
Comment (Schedulaar) [inline docs/02-common-problems.md]:
```suggestion
See [here](https://github.com/facebook/react-native/issues/21168#issuecomment-422700915) and [here](https://github.com/facebook/react-native/issues/20774) for more information.
```

File: docs/11-persistence.md [hunk 12]
```diff
@@ -0,0 +1,88 @@
...
+    "sorting_name": "Regensburg",
+    "events": true,
+    "push_notifications": true,
+    "extras": true
+  }
+}
+```
+
+The ''content'' is responsible to store the JSON data from the API.
+The ''resource-cache'' contains all the resources. Resources are files which are referenced in the HTML code and thumbnail of pages.
```
Comment (Schedulaar) [inline docs/11-persistence.md]:
```suggestion
The ''resource-cache'' contains all the resources. Resources are files which are either referenced in the HTML content or used as thumbnails of pages.
```

File: docs/12-design-changes.md [hunk 13]
```diff
@@ -0,0 +1,24 @@
+# Design Changes
+
+###### WIP Change: Improve "Offers"
+* Discussion on the next conference: Dissolve Extras category into Jobs, Apartment, Other (Other as category for extra wishes) → There is no icon and no wording that summarizes all extras, through the categorized presentation, the offers become more visible and more newcomers can benefit
```
Comment (Schedulaar) [inline docs/12-design-changes.md]:
```suggestion
* Discussion on the next conference: Break Extras up into Jobs, Apartment, Other (Other as category for extra wishes) → There is no icon and no wording that summarizes all extras, through the categorized presentation, the offers become more visible and more newcomers can benefit
```

> `Other` as category for extra wishes

Wishes of a certain municipality?

File: docs/12-design-changes.md [hunk 14]
```diff
@@ -0,0 +1,24 @@
...
+
+
+###### Design Change: Release of the react-native app
+The design of the react-native app will differ in a few points.
+* "Extras" will be changed to "Angebote" (eng. "Offers")
+* The tabs will get removed on Android and iOS
+* A Dashboard is added:
+ ![](figures/dashboard-design.jpg)
+ Note: The event preview is planned for a future design change.\
+ Note: The header should contain the Integreat logo xor the back button on iOS and Android.
```
Comment (Schedulaar) [inline docs/12-design-changes.md]: > Integreat logo

With `Integreat` lettering or only the square icon?

File: docs/12-design-changes.md (hunk 13 above)
Comment (maxammann) [reply docs/12-design-changes.md]: Result of the design workshop. Discussio will follow.

File: docs/01-debugging.md (hunk 3 above)
Comment (maxammann) [reply docs/01-debugging.md]: Using the correct term: reverse tunnel now

File: docs/12-design-changes.md (hunk 14 above)
Comment (Schedulaar) [reply docs/12-design-changes.md]: > Integreat logo

With `Integreat` lettering or only the square icon?

**Pull request merged**